browser-test/src/admin/admin_application_statuses.test.ts

```diff
@@ -359,6 +361,171 @@ test.describe('view program statuses', () => {
+        await adminPrograms.editProgram(programWithStatusesName)
+        await adminPrograms.editProgramBlock(
+          programWithStatusesName,
+          'block description',
+          ['Email'],
+        )
+        await adminPrograms.publishAllDrafts()
+        await logout(page)
+      })
+      test.beforeEach(async () => {
+        const {page} = ctx
+        // Logout before each test so we can start from the homescreen as an applicant
+        await logout(page)
+      })
+      test.afterAll(async () => {
+        const {page} = ctx
+        await disableFeatureFlag(
+          page,
+          'primary_applicant_info_questions_enabled',
+        )
+      })
+      test('email is displayed and sent for guest user that has answered the PAI email question', async () => {
+        const {page, applicantQuestions, adminPrograms} = ctx
+        const guestEmail = 'guestemail@example.com'
+
+        await applicantQuestions.applyProgram(programWithStatusesName)
+        await applicantQuestions.answerEmailQuestion(guestEmail)
+        await applicantQuestions.clickNext()
+        await applicantQuestions.submitFromReviewPage()
+        await logout(page)
+
+        await loginAsProgramAdmin(page)
+        await adminPrograms.viewApplications(programWithStatusesName)
+        await adminPrograms.viewApplicationForApplicant('Guest')
+
+        const emailsBefore = supportsEmailInspection()
+          ? await extractEmailsForRecipient(page, guestEmail)
+          : []
+
+        const modal =
+          await adminPrograms.setStatusOptionAndAwaitModal(emailStatusName)
+
+        const notifyCheckbox = await modal.$('input[type=checkbox]')
+        if (!notifyCheckbox) {
+          throw new Error('Expected a checkbox input')
+        }
+        expect(await notifyCheckbox.isChecked()).toBe(true)
+
+        expect(await modal.innerText()).toContain(
+          ' of this change at ' + guestEmail,
+        )
+        await adminPrograms.confirmStatusUpdateModal(modal)
+        expect(await adminPrograms.getStatusOption()).toBe(emailStatusName)
+        await adminPrograms.expectUpdateStatusToast()
+
+        if (supportsEmailInspection()) {
+          await adminPrograms.expectEmailSent(
+            emailsBefore.length,
+            guestEmail,
+            emailBody,
+            programWithStatusesName,
+          )
+        }
+      })
+      test('both email addresses are displayed and two emails are sent for a logged in user that has answered the PAI email question with a different email', async () => {
+        const {page, applicantQuestions, adminPrograms} = ctx
+        const otherTestUserEmail = 'other@example.com'
+
+        await loginAsTestUser(page)
+        await applicantQuestions.applyProgram(programWithStatusesName)
+        await applicantQuestions.answerEmailQuestion(otherTestUserEmail)
+        await applicantQuestions.clickNext()
+        await applicantQuestions.submitFromReviewPage()
+        await logout(page)
```

Comment:
```suggestion
        await test.step(`Apply to program ${programWithStatusesName} as test user`, async () => {
            await loginAsTestUser(page)
            await applicantQuestions.applyProgram(programWithStatusesName)
            await applicantQuestions.answerEmailQuestion(otherTestUserEmail)
            await applicantQuestions.clickNext()
            await applicantQuestions.submitFromReviewPage()
            await logout(page)
        })
```

I recommend you use [test.step](https://playwright.dev/docs/release-notes#-steps-api-with-teststep) to group the logical parts of your test.

Further down other steps for things like 
- view application as program admin
- do the thing with the modal
- verify emails sent

Notice the 'do the thing with the modal'. I wrote that because it's not super clear exactly what is going on just glancing at it. A good label will help us know what that steps does.

Reply:
Oh and steps can return values if that is ever needed